scipy/signal/_spectral_py.py

```diff
@@ -2007,10 +2060,11 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
     if sides == 'twosided':
         func = sp_fft.fft
     else:
-        result = result.real
+        if is_complex(result, xp=xp):
```

Comment:
Just above here, `torch` on GPU fails the proceedings paper benchmarks as well with:

<details>

```
Running:  python bench.py scipy torch_gpu
Traceback (most recent call last):
  File "/home/treddy/github_projects/scipy-2023-presentation/benchmarks/bench.py", line 130, in <module>
    main(benchmark, namespace)
  File "/home/treddy/github_projects/scipy-2023-presentation/benchmarks/bench.py", line 78, in main
    bench(namespace, print_times=False)
  File "/home/treddy/github_projects/scipy-2023-presentation/benchmarks/bench.py", line 119, in bench_scipy
    f, p = welch(x, nperseg=8)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/scipy/signal/_spectral_py.py", line 467, in welch
    freqs, Pxx = csd(x, x, fs=fs, window=window, nperseg=nperseg,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/scipy/signal/_spectral_py.py", line 610, in csd
    freqs, _, Pxy = _spectral_helper(x, y, fs, window, nperseg, noverlap,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/scipy/signal/_spectral_py.py", line 1963, in _spectral_helper
    result = _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/scipy/signal/_spectral_py.py", line 2057, in _fft_helper
    result = win * result
             ~~~~^~~~~~~~
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

</details>